src/network/frame_decoder.rs

```diff
@@ -345,4 +297,30 @@ impl<'a, 'b> FrameDecoder<'a, 'b> {
 
         Ok(frames)
     }
+
+    fn time_network_error(&self, frames: Vec<Frame>) -> Option<NetworkError> {
```

Comment:
Thanks for extracting this function to reduce duplication. Couple of notes:

1. There's no need for `time_network_error` to own the `frames` vector. Consider a reference to a frame slice (`frames: &[Frame]`). You'll need to update the call sites to be `self.time_network_error(&frames))`
2. I don't believe this keeps the same behavior (my fault for not having a test to stress this behavior 😄 ). If I'm not mistaken how you have it here -- only the last frame is looked at, where as the previous behavior would start at the last frame and work it's way to the front until a frame with an updated or new actor was found. I'm unsure if this distinction in behavior is actually important (as it could be that all frames contain a least one new or updated actor). But for the sake of this PR I'd like to keep the behavior the same.

Reply:
Updated.